usage/test-framework/src/main/java/org/apache/brooklyn/test/framework/SimpleCommandImpl.java

```diff
@@ -0,0 +1,177 @@
+     */
+    protected void handle(SimpleCommandDriver.Result result) {
+        LOG.debug("Result is {}\nwith output [\n{}\n] and error [\n{}\n]", new Object[] {
+                result.getExitCode(), shorten(result.getStdout()), shorten(result.getStderr())
+        });
+    }
+
+    private String shorten(String text) {
+        return Strings.maxlenWithEllipsis(text, A_LINE);
+    }
+
+    /**
+     * Does nothing in this class but gives sub-classes the opportunity to filter locations according to some criterion.
+     */
+    public Collection<? extends Location> filterLocations(Collection<? extends Location> locations) {
+        return locations;
+    }
+
+
+    @Override
+    public void init() {
+        super.init();
+        getLifecycleEffectorTasks().attachLifecycleEffectors(this);
+    }
+
+
+    protected void initDriver(MachineLocation machine) {
+        LOG.debug("Initializing simple command driver");
+        SimpleCommandDriver newDriver = doInitDriver(machine);
+        if (newDriver == null) {
+            throw new UnsupportedOperationException("cannot start "+this+" on "+machine+": no driver available");
+        }
+        driver = newDriver;
+    }
+
+    protected SimpleCommandDriver doInitDriver(MachineLocation machine) {
+        if (driver!=null) {
+            if (machine.equals(driver.getLocation())) {
+                return driver; //just reuse
+            } else {
+                LOG.warn("driver/location change is untested for {} at {}; changing driver and continuing", this, machine);
+                return newDriver(machine);
+            }
+        } else {
+            return newDriver(machine);
+        }
+    }
+
+    protected SimpleCommandDriver newDriver(MachineLocation machine) {
+        LOG.debug("Creating new simple command driver for {} from management context", machine);
+        EntityDriverManager entityDriverManager = getManagementContext().getEntityDriverManager();
+        return entityDriverManager.build(this, machine);
+    }
+
+    @Override
+    public void start(@EffectorParam(name = "locations") Collection<? extends Location> locations) {
+        this.locations = locations;
+        startOnLocations();
+    }
+
+    protected void startOnLocations() {
+        setExpectedState(this, STARTING);
+        int size = locations.size();
+        LOG.debug("Starting simple command at {} locations{}", size,
+                size > 0 ? " beginning " + locations.iterator().next() : "");
+        try {
+            execute(locations);
+            setUpAndRunState(true, RUNNING);
+
+        } catch (final Exception e) {
+            setUpAndRunState(false, ON_FIRE);
+            throw Exceptions.propagate(e);
+        }
+    }
+
+    private void execute(Collection<? extends Location> locations) {
+        SimpleCommandDriver.Result result = null;
+        String downloadUrl = getConfig(DOWNLOAD_URL);
+        if (Strings.isNonBlank(downloadUrl)) {
```

Comment:
I'd also log.warn if downloadUrl and defaultCommand are both set (because in that case we'll ignore their command).

Is DEFAULT_COMMAND the right name? It feels more like *the* command if the user is just supplying a simple command, rather than a URL to download a full script. Maybe this should just be "command"?